janus_core/helpers/utils.py

```diff
@@ -1,11 +1,147 @@
+    """
+
+    def __init__(
+        self,
+        struct: Atoms,
+        struct_name: Optional[str],
+        file_prefix: Optional[PathLike],
+        *additional,
+    ):
+        """
+        Provide mixin functions for standard filename handling.
+
+        Parameters
+        ----------
+        struct : Atoms
+            Structure from which to derive the default name if struct_name not provided.
+        struct_name : Optional[str]
+            Struct name to use.
+        file_prefix : Optional[PathLike]
+            Default prefix to use.
+        *additional
+            Components to add to file_prefix (joined by hyphens).
+        """
+        self.struct_name = self._get_default_struct_name(struct, struct_name)
+
+        self.file_prefix = Path(
+            self._get_default_prefix(file_prefix, self.struct_name, *additional)
+        )
+
+    @staticmethod
+    def _get_default_struct_name(struct: Atoms, struct_name: Optional[str]) -> str:
+        """
+        Determine the default struct name from the structure or provided struct_name.
+
+        Parameters
+        ----------
+        struct : Atoms
+            Structure of system.
+        struct_name : Optional[str]
+            Name of structure.
+
+        Returns
+        -------
+        str
+            Structure name.
+        """
+
+        if struct_name is not None:
+            return struct_name
+        return struct.get_chemical_formula()
+
+    @staticmethod
+    def _get_default_prefix(
+        file_prefix: Optional[PathLike], struct_name: str, *additional
+    ) -> PathLike:
+        """
+        Determine the default prefix from the structure name or provided file_prefix.
+
+        Parameters
+        ----------
+        file_prefix : str
+            Given file_prefix.
+        struct_name : str
+            Name of structure.
+        *additional
+            Components to add to file_prefix (joined by hyphens).
+
+        Returns
+        -------
+        str
+            File prefix.
+        """
+        if file_prefix is not None:
+            return str(file_prefix)
+        return "-".join((struct_name, *additional))
+
+    def _build_filename(
+        self,
+        suffix: str,
+        *additional,
+        filename: Optional[PathLike] = None,
+        prefix_override: Optional[str] = None,
+    ) -> PathLike:
+        """
+        Set filename using the file prefix and suffix if not specified otherwise.
+
+        Parameters
+        ----------
+        suffix : str
+            Default suffix to use if `filename` is not specified.
+        *additional
+            Extra components to add to suffix (joined with hyphens).
+        filename : Optional[PathLike]
+            Filename to use, if specified. Default is None.
+        prefix_override : Optional[str]
+            Replace file_prefix if not None.
+
+        Returns
+        -------
+        str
+            Filename specified, or default filename.
```

Comment:
```suggestion
    ) -> Path:
        """
        Set filename using the file prefix and suffix if not specified otherwise.

        Parameters
        ----------
        suffix : str
            Default suffix to use if `filename` is not specified.
        *additional
            Extra components to add to suffix (joined with hyphens).
        filename : Optional[PathLike]
            Filename to use, if specified. Default is None.
        prefix_override : Optional[str]
            Replace file_prefix if not None.

        Returns
        -------
        Path
            Filename specified, or default filename.
```